pkg/pillar/zedUpload/datastore_req.go

```diff
@@ -196,6 +196,22 @@ func (req *DronaRequest) updateOsize(size int64) {
 	req.objectSize = size
 }
 
+// Update the object size and chunk data
+func (req *DronaRequest) updateChunkData(size int64, chunk []byte) {
+	req.Lock()
+	defer req.Unlock()
+	req.objectSize = size
+	req.Adata = chunk
+}
+
+// clear the byte data in the req
+func (req *DronaRequest) cleanUpByteData() {
+	time.Sleep(CleanUpChunkDataTime)
```

Comment:
Why do you need to sleep for 10 ms? Is there a potential race condition?? 10ms will not fix a race; probably need some interlock if there is a race.